### ISSUE TITLE
esp-ieee802154: `ieee802154` vs `smoltcp` as the underlying implementation

### ISSUE THREAD

**Issue by ProfFan** [esp-rs/esp-hal#1559]
Just opening the discussion here. While the `ieee802154` recently has seen some activity it has been abandoned for 3+ years. While the `smoltcp`'s IEEE 802.15.4 support has been there for quite some time, and smoltcp has much more users.

Plus smoltcp's implementation is mostly zero-copy which should be beneficial for our application.

**Comment by MabezDev** [comment]
I don't think it really matters, we're currently using what works. We'd welcome a PR to change to smoltcp if its objectively better to do so, but I don't think its worth us tracking it here.
